ipn/local.go

```diff
@@ -812,7 +867,7 @@ func (b *LocalBackend) stateMachine() {
 
 func (b *LocalBackend) stopEngineAndWait() {
 	b.logf("stopEngineAndWait...")
-	b.e.Reconfig(&wgcfg.Config{}, nil, nil, false)
+	b.e.Reconfig(&wgcfg.Config{}, router.Settings{})
```

Comment:
(random place to comment, but...)

router.Settings is pretty large and pointer-heavy to be a value type. (https://github.com/golang/go/wiki/CodeReviewComments#receiver-type)

Any objections against changing to passing it around by pointer everywhere instead? Then this could just be nil.)

Reply:
No objection. Started out smaller and I embiggened it while splicing it out from wgcfg.Config. Adjusted to a pointer type, with nil handled gracefully within router to mean "a config that removes most tailscale customization from the network stack".